### PR TITLE
[CARBONDATA-1472] Optimize memory and fix nosort queries 

### PR DESCRIPTION
**Pull request by ravipesala**
1. Use UnsafeManager for dimension chunks as well to avoid leaks
2. Fix filters on nosort columns.
3. Optimize scanRDD

### CODE REVIEW

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/645/

**Comment by chenliang613** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/658/

**Comment by chenliang613** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/681/

**Comment by zzcclp** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/683/

**Comment by zzcclp** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/687/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/697/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/701/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/704/

File: core/src/main/java/org/apache/carbondata/core/datastore/chunk/store/impl/unsafe/UnsafeAbstractDimensionDataChunkStore.java [hunk 1]
```diff
@@ -68,10 +72,9 @@
    * @param numberOfRows   total number of rows
    */
   public UnsafeAbstractDimensionDataChunkStore(long totalSize, boolean isInvertedIdex,
-      int numberOfRows) {
+      int numberOfRows) throws MemoryException {
     // allocating the data page
-    this.dataPageMemoryBlock =
-        MemoryAllocatorFactory.INSATANCE.getMemoryAllocator().allocate(totalSize);
+    this.dataPageMemoryBlock = UnsafeMemoryManager.allocateMemoryWithRetry(taskId, totalSize);
```
[the remark this answers is not in the document]
Comment (ravipesala) [reply core/src/main/java/org/apache/carbondata/core/datastore/chunk/store/impl/unsafe/UnsafeAbstractDimensionDataChunkStore.java]: ok

File: core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelRangeLessThanEqualFilterExecuterImpl.java [hunk 2]
```diff
@@ -265,11 +265,15 @@ private BitSet getFilteredIndexes(DimensionColumnDataChunk dimensionColumnDataCh
       DirectDictionaryGenerator directDictionaryGenerator = DirectDictionaryKeyGeneratorFactory
```
[the remark this answers is not in the document]
Comment (ravipesala) [reply core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelRangeLessThanEqualFilterExecuterImpl.java]: ok

File: core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelRangeLessThanFiterExecuterImpl.java [hunk 3]
```diff
@@ -267,11 +267,15 @@ private BitSet getFilteredIndexes(DimensionColumnDataChunk dimensionColumnDataCh
       DirectDictionaryGenerator directDictionaryGenerator = DirectDictionaryKeyGeneratorFactory
```
[the remark this answers is not in the document]
Comment (ravipesala) [reply core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelRangeLessThanFiterExecuterImpl.java]: ok

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**